client/asset/eth/eth.go

```diff
@@ -3691,6 +3749,59 @@ func (w *assetWallet) checkFindRedemptions() {
 	}
 }
 
+// sumPendingTxs sums the expected incoming and outgoing values in unconfirmed
+// transactions stored in pendingTxs. Not used if the node is a mempoolFetcher.
+func (w *assetWallet) sumPendingTxs() (out, in uint64) {
+	w.tipMtx.RLock()
+	tip := w.currentTip.Number.Uint64()
+	w.tipMtx.RUnlock()
+
+	isETH := w.assetID == BipID
+	addPendingSend := func(pt *pendingTx) {
+		in += pt.in
+		if isETH {
+			if pt.assetID == BipID {
+				out += pt.out + pt.fees
+			} else {
+				out += pt.fees
+			}
+			return
+		}
+		// token
+		out += pt.out
+	}
+
+	w.pendingTxMtx.Lock()
+	defer w.pendingTxMtx.Unlock()
+	for txHash, pt := range w.pendingTxs {
+		if !isETH && pt.assetID != w.assetID {
+			continue
+		}
+		if pt.lastCheck == tip {
+			// Expect nothing has changed since our last check.
+			addPendingSend(pt)
+			continue
+		}
+		confs, err := w.node.transactionConfirmations(w.ctx, txHash)
+		if err != nil && !errors.Is(err, asset.CoinNotFoundError) {
+			w.log.Errorf("Error getting confirmations for pending tx %s: %v", txHash, err)
+			// DRAFT NOTE: Should I stop tracking this tx? I'd hate to end up in
```

Comment:
Can't hurt to stop tracking after a certain amount of blocks.

Reply:
It looks like if the coin is not found, this will check forever. So, if a tx was not mined for some reason, the balance will continue to be subtracted even if the tx has no chance of being mined. I had added a 2 minute timer to give up on txs and remove them from the cache.